datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/controller/VersionController.java

```diff
@@ -55,17 +67,25 @@
      */
     @SneakyThrows
     @GetMapping(value = "/versioned/{path:.*}", produces = APPLICATION_OCTET_STREAM_VALUE)
+    @ApiOperation("Read latest document from user's private space")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Document was successfully read"),
+            @ApiResponse(code = 403, message = "Access denied"),
+            @ApiResponse(code = 404, message = "Document not found")
+    })
     public void readVersionedDocument(@RequestHeader String user,
-                             @RequestHeader String password,
-                             @PathVariable String path,
-                             HttpServletResponse response) {
+                                      @RequestHeader String password,
+                                      @PathVariable String path,
+                                      HttpServletResponse response) {
         UserIDAuth userIDAuth = new UserIDAuth(new UserID(user), new ReadKeyPassword(password));
         PrivateResource resource = BasePrivateResource.forPrivate(path);
         ReadRequest<UserIDAuth, PrivateResource> request = ReadRequest.forPrivate(userIDAuth, resource);
         try (InputStream is = versionedDatasafeServices.latestPrivate().read(request);
              OutputStream os = response.getOutputStream()
         ) {
             StreamUtils.copy(is, os);
+        } catch (Exception e) {
```

Comment:
It is very generic wrapping - there can be credentials issues causing exception

Reply:
fixed